fuel-vm/src/tests/blockchain.rs

```diff
@@ -2261,3 +2261,444 @@ fn various_ldc_issues_poc() {
+
+    let blob_id = BlobId::compute(&program);
+
+    let tx_blob = TransactionBuilder::blob(BlobBody {
+        id: blob_id,
+        witness_index: 0,
+    })
+    .add_witness(program.into())
+    .maturity(maturity)
+    .add_random_fee_input()
+    .finalize()
+    .into_checked(height, &consensus_params)
+    .expect("failed to check tx");
+
+    client.blob(tx_blob).unwrap();
+
+    let load_blob = cmd;
+
+    let tx_deploy_loader =
+        TransactionBuilder::script(load_blob.into_iter().collect(), blob_id.to_vec())
+            .script_gas_limit(gas_limit)
+            .maturity(maturity)
+            .add_random_fee_input()
+            .finalize()
+            .into_checked(height, &consensus_params)
+            .expect("failed to check tx");
+
+    let receipts = client.transact(tx_deploy_loader);
+    if let Receipt::Panic { id: _, reason, .. } = receipts.first().expect("No receipt") {
+        assert_eq!(
+            &expected_reason,
+            reason.reason(),
+            "Expected {}, found {}",
+            expected_reason,
+            reason.reason()
+        );
+    } else {
+        panic!("Script should have panicked before logging");
+    }
+}
+
+#[test]
+fn ldcv2__fails_with_nonempty_stack() {
+    let (load_blob, _) = script_with_data_offset!(
+        data_offset,
+        vec![
+            op::movi(0x10, data_offset as Immediate18),
+            op::cfei(0x1), // sp += 1
+            op::ldc(0x10, RegId::ZERO, RegId::ONE, 1),
+        ],
+        TxParameters::DEFAULT.tx_offset()
+    );
+
+    ldcv2_reason_helper(load_blob, ExpectedUnallocatedStack);
+}
+
+#[test]
+fn ldcv2__fails_when_mem_offset_is_above_reg_hp() {
+    // Then deploy another contract that attempts to read the first one
+
+    let (load_blob, _) = script_with_data_offset!(
+        data_offset,
+        vec![
+            op::movi(0x10, data_offset as Immediate18),
+            op::ldc(0x10, RegId::ZERO, RegId::HP, 1),
+        ],
+        TxParameters::DEFAULT.tx_offset()
+    );
+
+    ldcv2_reason_helper(load_blob, MemoryOverflow);
+}
+
+#[test]
+fn ldcv2__fails_when_blob_size_overflows() {
+    ldcv2_reason_helper(
+        vec![
+            op::not(0x20, RegId::ZERO),
+            op::ldc(RegId::HP, RegId::ZERO, 0x20, 1),
+        ],
+        MemoryOverflow,
+    );
+}
+
+#[test]
+fn ldcv2__fails_wehn_blob_id_ends_beyond_max_ram() {
+    // Then deploy another contract that attempts to read the first one
+    let reg_a = 0x20;
+    let reg_b = 0x21;
+
+    // cover blob_id_end beyond max ram
+    let load_blob = vec![
+        op::move_(reg_a, RegId::HP),           // r[a] := $hp
+        op::xor(reg_b, reg_b, reg_b),          // r[b] := 0
+        op::ori(reg_b, reg_b, 12),             // r[b] += 12 (will be padded to 16)
+        op::ldc(reg_a, RegId::ZERO, reg_b, 1), // Load first two words from the contract
+    ];
+
+    ldcv2_reason_helper(load_blob, MemoryOverflow);
+}
+
+#[test]
+fn ldcv2_fails_when_blob_not_in_inputs() {
```

Comment:
We don't require the blob to be in inputs. What are we testing here?

Reply:
It's test for nonexistent blob, but named incorrectly. I'll fix.

Reply:
02df199a